Mathlib/Data/Set/Countable.lean

```diff
@@ -23,23 +37,22 @@ variable {α : Type u} {β : Type v} {γ : Type w} {ι : Sort x}
 
 namespace Set
 
-/-- A set is countable if there exists an encoding of the set into the natural numbers.
-An encoding is an injection with a partial inverse, which can be viewed as a
-constructive analogue of countability. (For the most part, theorems about
-`Countable` will be classical and `Encodable` will be constructive.)
+/-- A set `s`  is countable if the corresponding subtype is countable,
+i.e., there exists an injective map `f : s → ℕ`.
+
+Note that this is an abbreviation, so `hs : Set.Countable s` in the proof context
+is the same as an instance `Countable s`.
+For a constructive version, see `Encodable`.
 -/
-protected def Countable (s : Set α) : Prop :=
-  Nonempty (Encodable s)
+protected abbrev Countable (s : Set α) : Prop := Countable s
```

Comment:
Does this mean that `[S.Countable]` is now usable as a typeclass argument?

Reply:
Yes, and it means the same as `[Countable s]`.

Reply:
I'd be tempted to use the safer
```suggestion
structure Countable (s : Set α) : Prop where
  countable_coe : Countable s
```
or if that's really objectionable,
```suggestion
protected def Countable (s : Set α) : Prop := Countable s
```

Reply:
What exactly are you afraid of? I deleted `have := fun a ↦ (h a).to_subtype` or `have := h.to_subtype` here and there because it's an `abbrev` now. I can restore it but I need a reason.

Reply:
!bench

Reply:
UPD: I see that you answered on Zulip. Let's discuss there.